packages/components/pager/src/pager.js

```diff
@@ -0,0 +1,150 @@
+import css from './pager.css';
+import { customEventNames } from './utils';
+
+customElementDefineHelper(
+	'ts-pager',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				pages: { type: Number, reflect: true, attribute: true },
+				page: { type: Number, reflect: true, attribute: true },
+				max: { type: Number, reflect: true, attribute: true }
+			};
+		}
+
+		constructor() {
+			super();
+			this.page = 1;
+			this.max = 5;
+		}
+
+		attributeChangedCallback(name, oldVal, newVal) {
+			super.attributeChangedCallback(name, oldVal, newVal);
+			if (name === 'pages' || name === 'max') {
+				this.resetRange();
+			}
+			if (name === 'page') {
+				this.dispatchCustomEvent(customEventNames.SELECT, { oldVal, newVal });
+			}
+		}
+
+		resetRange() {
+			this.start = this.page;
+			this.end = Math.min(this.start + Math.min(this.pages, this.max) - 1, this.pages);
+		}
+
+		get toStart() {
+			const disabled = this.start === 1;
+			return html`
+				<li>
+					<button ?disabled=${disabled} @click="${this.handleToStart}">
+						|&lt;
+					</button>
+				</li>
+			`;
+		}
+
+		handleToStart() {
+			this.start = 1;
+			this.page = 1;
+			this.end = Math.min(this.start + Math.min(this.pages, this.max) - 1, this.pages);
+		}
+
+		get toEnd() {
+			const disabled = this.end === this.pages;
+			return html`
+				<li>
+					<button ?disabled=${disabled} @click="${this.handleToEnd}">
+						&gt;|
+					</button>
+				</li>
+			`;
+		}
+
+		handleToEnd() {
+			this.page = this.pages;
+			this.end = this.pages;
+			this.start = this.end - Math.min(this.pages, this.max) + 1;
+		}
+
+		get prev() {
+			const disabled = this.page === this.start && this.start === 1;
+			return html`
+				<li>
+					<button ?disabled=${disabled} @click="${this.handleToPrev}">
+						&lt;&lt;
+					</button>
+				</li>
+			`;
+		}
+
+		handleToPrev() {
+			if (this.page === 0) {
+				return;
+			}
+			if (this.page === this.start) {
+				this.start--;
+				this.end--;
+			}
+			this.page--;
+		}
+
+		get next() {
+			const disabled = this.page === this.end && this.end === this.pages;
+			return html`
+				<li>
+					<button ?disabled=${disabled} @click="${this.handleToNext}">
+						&gt;&gt;
+					</button>
+				</li>
+			`;
+		}
+
+		handleToNext() {
+			if (this.page === this.pages) {
+				return;
+			}
+			if (this.page === this.end) {
+				this.start++;
+				this.end++;
+			}
+			this.page++;
+		}
+
+		get numbers() {
+			let res = [];
+			for (let i = this.start; i <= this.end; i++) {
+				const disabled = this.page === i;
+				res.push(
+					html`
+						<li>
+							<button data-value="${i}" ?disabled=${disabled} @click="${this.handleNumber}">
+								${i}
+							</button>
+						</li>
+					`
+				);
+			}
+			return html`
+				${res}
+			`;
+		}
+
+		handleNumber(e) {
+			this.page = parseInt(e.target.getAttribute('data-value'));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add the base to this "parseInt" call. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AS2427)